src/main/java/com/microsoft/sqlserver/jdbc/SQLServerStatement.java

```diff
@@ -1601,9 +1601,15 @@ boolean onDone(TDSReader tdsReader) throws SQLServerException {
                         if (null != procedureName)
                             return false;
 
+                        //For Insert, we must fetch additional TDS_DONE token that comes with the actual update count
+                        if (doneToken.cmdIsInsert() &&  (-1 != doneToken.getUpdateCount()) && EXECUTE == executeMethod) {
```

Comment:
I would prefer to use the existing method.
```suggestion
                        if (StreamDone.CMD_INSERT == doneToken.getCurCmd() &&  (-1 != doneToken.getUpdateCount()) && EXECUTE == executeMethod) {
```
and then remove the added method `cmdIsInsert()`.